src/components/zoneddatetime.rs

```diff
@@ -358,6 +455,40 @@ impl ZonedDateTime {
     }
 }
 
+// ==== HoursInDay accessor method implementation ====
+
+impl ZonedDateTime {
+    #[cfg(feature = "experimental")]
+    pub fn hours_in_day(&self) -> TemporalResult<u8> {
+        let provider = TZ_PROVIDER
+            .lock()
+            .map_err(|_| TemporalError::general("Unable to acquire lock"))?;
+        self.hours_in_day_with_provider(provider.deref())
+    }
+
+    pub fn hours_in_day_with_provider(&self, provider: &impl TzProvider) -> TemporalResult<u8> {
```

Comment:
Thought: we should probably change the provider argument to `&(impl TzProvider + ?Sized)`, since the current type doesn't allow passing trait objects.

Reply:
Hmmm, true. I'll update in a follow up.